### PR TITLE
Allow open to work with 'from ...' block commands

### PR DESCRIPTION
**Pull request by neosam**
# Description

The open command failed when there was a 'from (file extension)' command which was a BlockCommand.  It tried to call the run method of the BlockCommand which is not able to evaluate itself.  Open now checks if the from-command has a block assigned and evaluates it if it does.  Otherwise it will call it's run method.

I also changed the run method in BlockCommand: Instead of panicking, the run method now returns an error to prevent
nushell from unexpected termination.

Fixes issue #5046 

# Tests

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by jntrnr** [comment]
@neosam nice! love it. Can you add a test as well?

**Comment by neosam** [comment]
@jntrnr Thank you!  Sure!  I'll look at it in a few hours.

**Comment by neosam** [comment]
I added a test case where a semicolon separated file is parsed.  I'm not sure if this is the expected way a test should look like in nu, though.

File: crates/nu-command/tests/commands/open.rs [hunk 1]
```diff
@@ -260,3 +260,21 @@ fn open_dir_is_ls() {
...
     })
 }
+
+
+#[test]
+fn test_open_block_command() {
+    let actual = nu!(
+        cwd: "tests/fixtures/formats",
+        r#"
+            def "from ssv" [] { lines | split column ";" }
```
Comment (jntrnr) [inline crates/nu-command/tests/commands/open.rs:269]: `from ssv` is already a command inside of nushell. Just to make the test clearer, you might want to choose one that is specific to the test and not part of nushell

Comment (neosam) [reply crates/nu-command/tests/commands/open.rs]: Oh, I wasn't aware that ssv actually exists.  I've changed it and I'm pretty sure the new custom file extension 'blockcommandparser' with ',|,' as separator doesn´t exist.

And sorry, I forgot to run the checks, especially 'cargo fmt' on my last commit.

**Comment by jntrnr** [comment]
nice!

**Pull request merged**